openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/changeEvent/AbstractEventConsumer.java

```diff
@@ -0,0 +1,247 @@
+    List<ChangeEvent> filteredEvents = new ArrayList<>();
+    for (ChangeEvent event : events) {
+      boolean triggerChangeEvent =
+          AlertUtil.shouldTriggerAlert(
+              event.getEntityType(), eventSubscription.getFilteringRules());
+
+      // Evaluate ChangeEvent Alert Filtering
+      if (eventSubscription.getFilteringRules() != null
+          && !AlertUtil.evaluateAlertConditions(
+              event, eventSubscription.getFilteringRules().getRules())) {
+        triggerChangeEvent = false;
+      }
+
+      if (triggerChangeEvent) {
+        // Ignore the event since change description is null
+        if (event.getChangeDescription() != null) {
+          filteredEvents.add(event);
+        } else {
+          LOG.info(
+              "Email Publisher Event Will be Ignored Since Change Description is null. Received Event: {}",
+              event);
+        }
+      }
+    }
+
+    try {
+      sendAlert(filteredEvents);
+      return true;
+    } catch (RetriableException ex) {
+      setNextBackOff();
+      LOG.error(
+          "Failed to publish event in batch {} due to {}, will try again in {} ms",
+          filteredEvents,
+          ex,
+          currentBackoffTime);
+      Thread.sleep(currentBackoffTime);
+    } catch (Exception e) {
+      LOG.error("[AbstractAlertPublisher] error {}", e.getMessage(), e);
+    }
+    return false;
+  }
+
+  @Override
+  public void commitOffset(JobExecutionContext jobExecutionContext, int offset) {
+    EventSubscriptionOffset eventSubscriptionOffset =
+        new EventSubscriptionOffset().withOffset(offset).withTimestamp(System.currentTimeMillis());
+    Entity.getCollectionDAO()
+        .eventSubscriptionDAO()
+        .upsertSubscriberOffset(
+            eventSubscription.getId().toString(),
+            OFFSET_EXTENSION,
+            "eventSubscriptionOffset",
+            JsonUtils.pojoToJson(eventSubscriptionOffset));
+
+    // Update the Job Data Map with the latest offset
+    jobExecutionContext.getJobDetail().getJobDataMap().put(ALERT_OFFSET_KEY, offset);
+  }
+
+  public synchronized void setErrorStatus(Long attemptTime, Integer statusCode, String reason) {
+    setStatus(FAILED, attemptTime, statusCode, reason, null);
+  }
+
+  public synchronized void setAwaitingRetry(Long attemptTime, int statusCode, String reason) {
+    setStatus(AWAITING_RETRY, attemptTime, statusCode, reason, attemptTime + currentBackoffTime);
+  }
+
+  public synchronized void setSuccessStatus(Long updateTime) {
+    SubscriptionStatus subStatus =
+        AlertUtil.buildSubscriptionStatus(
+            ACTIVE, updateTime, null, null, null, updateTime, updateTime);
+    eventSubscription.setStatusDetails(subStatus);
+  }
+
+  protected synchronized void setStatus(
+      SubscriptionStatus.Status status,
+      Long attemptTime,
+      Integer statusCode,
+      String reason,
+      Long timestamp) {
+    SubscriptionStatus subStatus =
```

Comment:
Do we need to be synchronized for these methods

Reply:
no